IV Objetivo 2 [IV-21-22]

eantoniocalo18 opened the pull request:
Entrega del objetivo 2 incluyendo:
- Elección del lenguaje de programación 
- Actualización de .gitignore para python
- Se incluyen #29 y #30 
- Se ha implementado las clases Horario y Proyecto y los problemas relacionados durante el proceso se han reflejado en #31 y #32

eantoniocalo18 committed:
Elección de lenguaje de programación Python

Actualizando .gitignore para el uso de python por su facilidad, legibilidad y escalabilidad.
Closes #28

eantoniocalo18 committed:
Representación clase Horario y Proyecto

Representación de las Clases Horario y Proyecto para avanzar en M0.
- Clase Horario: 
  - Contiene los parámetros id, usuario_relacionado, horas_semanales y dias_trabajados
  - Contiene los métodos: horas_trabajadas, add_dia, get_dias_trabajados, get_horas_restantes y to_CSV.
Los métodos get_horas_restantes y to_CSV permiten: closes morevi#29 y closes morevi#30 respectivamente

-Clase Proyecto: 
   - Contiene los parámetros id, nombre, planificación y usuarios_asignados
   -  Contiene los métodos: is_finalizado y horas_restantes

Con la creación de dichas clases se ha determinado la necesidad de representar un Usuario y se ha creado el issue correspondiente

eantoniocalo18 committed:
Creación de la clase Usuario

Para avanzar con M0, además de solucionar el issue #31 solo con la creación.
Por otra parte, los métodos creados: crear_horario, actualizar_horario y confirmar_horario permiten:
-closes #15
-closes #16
-closes #17

eantoniocalo18 committed:
Creación clase día

Creación de la clase día para solventar closes #32 y seguir avanzando con M0.
Esta clase contendrá la información relativa a las horas y la fecha de un día de trabajo por parte del usuario.

eantoniocalo18 committed:
Create vi.yaml

Creando vi.yaml para completar el hito del Objetivo 2

eantoniocalo18 committed:
Merge pull request #1 from eantoniocalo18/IV-Objetivo-2-IV-21-22]

IV Objetivo 2 [IV-21-22]

eantoniocalo18 commented:
Cuando puedas @morevi, enlaza este PR al Milestone 0 y los issues correspondientes

morevi commented on src/proyecto.py:
```diff
@@ -0,0 +1,46 @@
+        "Método que devolverá true si el proyecto ha sido terminado"
+        pass
+    def horas_restantes():
+        "Método que devolverá true si el proyecto ha sido terminado"
```
El comentario es erróneo

morevi commented on src/usuario.py:
```diff
@@ -0,0 +1,54 @@
+
+
+
+class usuario:
```
Nombres de clases deben comenzar por Mayuscula

morevi commented on src/usuario.py:
```diff
@@ -0,0 +1,54 @@
+
+class usuario:
+    """
+    Clase utilizada para representar un Proyecto.
```
Clase usuario/empleado, no proyecto

morevi reviewed (approved):
Me gusta la idea de tambien poder representar los días.
A partir de aquí se podrá avanzar a buen ritmo.

Hay algunos errores, pero nada grave que no se pueda arreglar en un commit de typos.
👍🏼

eantoniocalo18 committed:
Arreglando fallos typos indicados por @morevi. @JJ listo para revisión

morevi reviewed (approved):
Perfecto! 👍🏼

eantoniocalo18 commented:
@JJ Listo para revisión

JJ commented on src/dia.py:
```diff
@@ -0,0 +1,49 @@
+#----------------------------------------------------------------------------
+# Este archivo dia.py contiene la implementaciÃ³n de la clase dia.
```
???

JJ replied:
Sigue estando el error.

JJ commented on src/dia.py:
```diff
@@ -0,0 +1,49 @@
+        self.hora_fin = hora_fin
+        self.horas_descanso = horas_descanso
+
+    def calcular_horas():
```
¿Esto no sería hora_fin - hora_inicio - horas_descanso? ¿Por qué no es un atributo de clase?

eantoniocalo18 replied:
Había pensado declararlo como atributo de la clase pero he pensado que es más útil que puedas ir añadiéndole los datos progresivamente, es decir, si tú quieres añadir solo la hora de entrada y cuando salgas la hora de salida y llamar al método calcular_horas  una vez se haya terminado todo el horario completo.

JJ replied:
Pienses lo que pienses, tienes que ponerlo en un issue y expresar el problema que hay. Todo lo que  comentas, ¿en qué HU/issue lo dice?

JJ replied:
Sigues sin responder esto.

JJ commented on src/horario.py:
```diff
@@ -0,0 +1,58 @@
+    """
+    def __init__(self, id, usuario_relacionado,horas_semanales, dias_trabajados):
+        self.id = id
+        self.nombre = usuario_relacionado
```
Si el usuario tiene un id, ¿por qué se usa el nombre?

eantoniocalo18 replied:
Porque el id es un identificador único y los demás datos del usuario sirven para ser reflejados a la hora de subir sus horarios en un proyecto (pj Participantes del proyecto: Nombre-Apellidos)

eantoniocalo18 replied:
Tiene razón, es necesario que el atributo que represente la asociación entre las clases Usuario y Horario sea único, se ha representado el problema en #35

JJ commented on src/usuario.py:
```diff
@@ -0,0 +1,54 @@
+    Publicos
+    id: str
+        uuid o id que identifica cada usuario único
+    email : str
```
¿En qué ayuda esto a la lógica de negocio?

eantoniocalo18 replied:
Tiene razón, se ha creado #34 representando el problema

JJ replied:
El "problema" no es lo que yo te diga aquí, sino que tienes que crear un issue con los diferentes problemas planteados. Y el email ha desaparecido, pero el comentario sigue.

JJ reviewed (changes requested):
Repetidamente se ha dicho que un issue no es "crear no sé qué" y se soluciona con "esto está creado". Por favor, revisa la lista de comprobación y, si es necesario, pégala en el cuerpo del PR.

eantoniocalo18 committed:
Arreglando fallos y modificando los issues

Se han solucionado los fallos representados en #34 y #35.
Por otro lado, se ha tratado de dar una estructura a los issues #31 y #32 de manera que representen un problema

eantoniocalo18 commented:
@JJ Listo para revisión

JJ commented on src/dia.py:
```diff
@@ -0,0 +1,49 @@
+   
+    Publicos
+    id: str
+        uuid o id que identifica cada día único
```
¿En serio necesitáis un ID para representar un día? ¿No será suficiene con la fecha?

JJ commented on src/horario.py:
```diff
@@ -0,0 +1,58 @@
+    def __init__(self, id, usuario_relacionado,horas_semanales, dias_trabajados):
+        self.id = id
+        self.id_usuario = usuario_relacionado
+        self.horas_semanales = horas_semanales
```
¿Un horario tiene horas semanales y días trabajados? ¿Un usuario puede tener varios horarios? Si no, ¿por qué necesitas un id?

JJ replied:
Y esto

JJ reviewed (changes requested):
Aparte de los errores señalados y que realmente no se han solucionado, has modificado en un solo commit cosas de dos issues que has abierto ex profeso más otros dos de antes, y que siguen sin representar ningún problema. 
La mayoría del código sigue teniendo como último commit "creación clase xxx" en vez de indicar como resuelve un problema formulado en un issue. Por ejemplo, "clase Dia" no está claro que resuelva ningún problema ni sirva para nada en la lógica de negocio.

eantoniocalo18 committed:
Eliminando comentario e-mail

Se ha eliminado el comentario del e-mail y se ha tomado la elección de identificar el usuario con un id único.
Con eso se soluciona closes #34 y se avanza en M0

eantoniocalo18 committed:
Eliminando el ID día

Se ha eliminado el atributo id de la clase Día porque se ha decidido utilizar el atributo fecha ya que un usuario no va a trabajar dos veces en el mismo día.
Con esto closes #39 y se avanza en M0

eantoniocalo18 commented:
@JJ Listo para revisión

eantoniocalo18 commented:
> Aparte de los errores señalados y que realmente no se han solucionado, has modificado en un solo commit cosas de dos issues que has abierto ex profeso más otros dos de antes, y que siguen sin representar ningún problema. La mayoría del código sigue teniendo como último commit "creación clase xxx" en vez de indicar como resuelve un problema formulado en un issue. Por ejemplo, "clase Dia" no está claro que resuelva ningún problema ni sirva para nada en la lógica de negocio.

No sé cómo solucionar la parte de que "la mayoría del código sigue teniendo como último commit "creacion de xxx" ". He modificado el issue #32  en el que se intenta representar de una manera más clara el problema que ha surgido y cuya solución viene dada con la creación de la clase Dia en el commit b1324f10ba0b99aaf4e337d395f8b02483526052

JJ commented:
> > Aparte de los errores señalados y que realmente no se han solucionado, has modificado en un solo commit cosas de dos issues que has abierto ex profeso más otros dos de antes, y que siguen sin representar ningún problema. La mayoría del código sigue teniendo como último commit "creación clase xxx" en vez de indicar como resuelve un problema formulado en un issue. Por ejemplo, "clase Dia" no está claro que resuelva ningún problema ni sirva para nada en la lógica de negocio.
> 
> No sé cómo solucionar la parte de que "la mayoría del código sigue teniendo como último commit "creacion de xxx" ". He modificado el issue #32 en el que se intenta representar de una manera más clara el problema que ha surgido y cuya solución viene dada con la creación de la clase Dia en el commit [b1324f1](https://github.com/morevi/jobcontrol/commit/b1324f10ba0b99aaf4e337d395f8b02483526052)

Si no has escrito el código solucionando ningún problema, lo mejor que haces es borrarlo y empezar de nuevo. No se trata de que enlacéis cosas, se trata de que sigáis buenas prácticas para que el código resuelva el problema que tiene que resolver.

JJ commented on src/proyecto.py:
```diff
@@ -0,0 +1,46 @@
+        self.planificacion = planificacion
+        self.usuarios_asignados = usuarios_asignados
+
+    def is_finalizado():
```
¿Donde dice que hay que hacer esto y no, por ejemplo, borrar el proyecto?

JJ commented on src/proyecto.py:
```diff
@@ -0,0 +1,46 @@
+    def is_finalizado():
+        "Método que devolverá true si el proyecto ha sido terminado"
+        pass
+    def horas_restantes():
```
¿Cómo se supone que va a calcular esto? ¿En base a qué? ¿La planificación, dónde aparece? ¿Ves lo que pasa si no planteas ningún problema y pones cosas sin responder a ningún criterio? @morevi, ¿no tienes nada que decir sobre esto?

JJ commented on src/usuario.py:
```diff
@@ -0,0 +1,51 @@
+    def actualizar_horario():
+        "Método que modificará los datos de un Horario"
+        pass
+    def confirmar_horario():
```
Esto está creado en un commit que dice "creación clase usuario". Ni se sabe qué hace, ni qué debe hacer, ni cómo va a confirmar el horario (o no) en base al nombre y apellidos del usuario. Además, cada clase debe hacer una cosa y sólo una cosa. Si representa a un usuario, representa a un usuario.

JJ reviewed (changes requested):
Sigues sin responder a los múltiples comentarios, y sigues desarrollando la mayor parte del código sin indicar qué problema se resuelve ni por qué usas esas funciones y atributos.

eantoniocalo18 commented:
Se va a cerrar este PR, para crear uno desde 0, con una mayor limpieza y con los contenidos mejor enfocados
